engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -535,8 +534,12 @@ public void runLocal(AppFactory appConfig) throws Exception
 
   public URLClassLoader loadDependencies()
   {
-    URLClassLoader cl = URLClassLoader.newInstance(launchDependencies.toArray(new URL[launchDependencies.size()]));
+    URL[] dependencies = launchDependencies.toArray(new URL[launchDependencies.size()]);
+
+    ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
I'm not suggesting to merge the PR just yet, I wan't to make sure we actually resolve the issue before applying optimizations. For example, the PR needs to address removing the prior context class loader before the next application is launched.